Getters for gas consumption

oischinger opened the pull request:
Gas heaters are providing some statistics under "heating.gas.consumption.heating".
This change makes the data available either as an array of values (e.g. last 8 days, last 12 months, etc) or as a single value for today, this week, this month or this year.

oischinger committed:
Getters for gas consumption

Gas heaters are providing some statistics under "heating.gas.consumption.heating".
This change makes the data available either as an array of values (e.g. last 8 days, last 12 months, etc) or as a single value for today, this week, this month or this year.

somm15 commented:
Do you mind if I change 
getGasConsumptionDay() to getGasConsumptionDays() or getGasConsumptionPerDay()
I would rather use the name getGasConsumptionDay for getGasConsumptionDay(day)
It would be more coherent for getPrograms() for instance.

somm15 commented:
And by the way, thanks!
It looks like my Vitodens 222W does not support this feature, so I couldn't test this API endpoint.

somm15 reviewed (changes requested):
Could you please change naming when it return multiple values ?
e.g.  def getGasConsumptionMonth(self) -->  def getGasConsumptionMonths(self)

oischinger commented:
Actually I found out there is also a consumption for dhw. Let me add this and change the API as you requested. 
Thanks for this nice lib by the way.

oischinger committed:
Add DomesticHotWater statistics and rename heating statistic functions

oischinger commented:
Hi @somm15 
I added the getters for DHW and renamed the existing ones to have "Heating" in the name.
Also getGasConsumptionDay() is now getGasConsumptionDays() and so on.

Unfortunately this leads to a lot of functions. Please let me know if you prefer a different API. I'm not a python expert, so maybe this code is not really python style

oischinger commented:
oh, that was fast ;)
Thanks

oischinger commented:
Ups actually too fast. I just noticed that it should be "heating.gas.consumption.dhw" instead of "heating.gas.consumption.heating" in the dhw-Methods.
Sorry, copy paste error!

somm15 commented:
Do you want to make another commit?
(that indeed a lot of function, we could split in 3 objects
- Viessmann
|--> heating
|--> hot water
